tests/test_compile_error.rs

```diff
@@ -65,3 +65,14 @@ fn test_compile_errors() {
     t.compile_fail("tests/ui/invalid_cancel_handle.rs");
     t.pass("tests/ui/pymodule_missing_docs.rs");
 }
+
+#[cfg(not(target_arch = "wasm32"))] // Not possible to invoke compiler from wasm
+#[test]
+fn test_diagnostics() {
```

Comment:
Now that we're not gating the above, do we need this separate test?

Reply:
Usually my hope for all these tests is to have a good error message, so I think it's simpler to combine. 😁

Reply:
Ok :)